### DISCUSSION TITLE
How to access id values from inside widgets?

### DISCUSSION THREAD

**Discussion by White-Rabbit-Scientific** [beeware/toga#1667] [Q&A]
```
def callback_dec_prog_bar(pb, widget):
    pb.value -= 5  # <<< WORKS
    print(widget.children[0].id) # <<< THIS PRINTS "hello" BUT CAN THE "id" TEXT BE ACCESSED WITHOUT USING THE INDEX?


btn_txtbox = toga.Box(
    children=[
        toga.Button(id="hello", text="text", style=button_style, on_press=lambda _ : callback_dec_prog_bar(progbar, btn_txtbox))
        toga.Button2(.....)
        toga.Button3(.....) etc.
    ]
)
```

**Comment by freakboy3742** [comment]
I'm not exactly sure what you're asking. You can access the id attribute anywhere that you have a widget; `id` is an attribute of a widget object. However, you need to have an actual widget in mind - Python can't work out magically what widget you want the ID of. So - if you want the ID of a *specific* widget, you need to provide a reference to that widget.

As best as I can work out, you're trying to get the ID of the button that was pressed... which *is* provided to you, but you're throwing that information away. It's the first argument passed to the handler method when it's invoked - but you're throwing that value away when you wrap your handler in a lambda.
